capstone/src/test/java/com/google/sps/LessonTest.java

```diff
@@ -28,4 +49,150 @@ public void setUp() {
+        Assert.assertEquals(form.getUrl(), LessonTest.TEST_URL);
+    }
+
+    @Test
+    public void deserializeJsonVideoTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TYPE_PROPERTY_KEY, Lesson.TYPE_VIDEO);
+        jsonObject.addProperty(Lesson.TITLE_PROPERTY_KEY, LessonTest.TEST_TITLE);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Video.URL_PROPERTY_KEY, LessonTest.TEST_URL);
+
+        Lesson lesson = Lesson.deserializeJson(jsonObject.toString());
+
+        Assert.assertEquals(lesson.getType(), Lesson.TYPE_VIDEO);
+        Assert.assertEquals(lesson.getTitle(), LessonTest.TEST_TITLE);
+        Assert.assertEquals(lesson.getDescription(), LessonTest.TEST_DESCRIPTION);
+        Assert.assertEquals(lesson.getIsDraft(), false);
+        Assert.assertTrue(lesson instanceof Video);
+        Video video = (Video) lesson; 
+        Assert.assertEquals(video.getUrl(), LessonTest.TEST_URL);
+    }
+
+    @Test
+    public void deserializeJsonImageTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TYPE_PROPERTY_KEY, Lesson.TYPE_IMAGE);
+        jsonObject.addProperty(Lesson.TITLE_PROPERTY_KEY, LessonTest.TEST_TITLE);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Image.URL_PROPERTY_KEY, LessonTest.TEST_URL);
+
+        Lesson lesson = Lesson.deserializeJson(jsonObject.toString());
+
+        Assert.assertEquals(lesson.getType(), Lesson.TYPE_IMAGE);
+        Assert.assertEquals(lesson.getTitle(), LessonTest.TEST_TITLE);
+        Assert.assertEquals(lesson.getDescription(), LessonTest.TEST_DESCRIPTION);
+        Assert.assertEquals(lesson.getIsDraft(), false);
+        Assert.assertTrue(lesson instanceof Image);
+        Image image = (Image) lesson; 
+        Assert.assertEquals(image.getUrl(), LessonTest.TEST_URL);
+    }
+
+    @Test
+    public void deserializeJsonContentTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TYPE_PROPERTY_KEY, Lesson.TYPE_CONTENT);
+        jsonObject.addProperty(Lesson.TITLE_PROPERTY_KEY, LessonTest.TEST_TITLE);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Content.CONTENT_PROPERTY_KEY, LessonTest.TEST_CONTENT);
+
+        JsonArray jsonArray = new JsonArray();
+        jsonArray.add(LessonTest.TEST_URL);
+        jsonArray.add(LessonTest.TEST_EDIT_URL);
+        jsonObject.add(Content.URLS_PROPERTY_KEY, (JsonElement) jsonArray);
+
+        ArrayList<String> testList = new ArrayList<>();
+        Iterator<JsonElement> listIterator = jsonArray.iterator();
+        while(listIterator.hasNext()){
+            testList.add(listIterator.next().getAsString());
+        }
```

Comment:
I found this to be the best way to get replicate a collection of URLs. I could put the construction of the jsonarray in setUp() but decided to leave it here since this is the only test that use it (can add to setUp() in future PR when we test more).

Reply:
Also is there another "better" way to get a jsonArray to an array of type String to compare in assertArrayEquals()?

Reply:
I dont know that there is one that is much "better". You could use streams and that might be _slightly_ more readable, but if you are not familiar with the API, then its not worth changing. Java is known for being a somewhat verbose language, so... it taking this much code is somewhat expected.

I think its the right call to leave it in this test, and even potentially copy-paste it if multiple tests use it. setUp is more meant for things which are completely irrelevant to the test, like... idk mocking out some dependencies or something

Reply:
Alrighty. Got it